File: skills/html-generator/scripts/html-gen.py
#!/usr/bin/env python3
"""HTML Generator - 调用 Seedream API 生成图片"""

import requests
import json
import sys
import os

# 配置 - 使用稳定版模型
SEEDREAM_URL = "https://ark.cn-beijing.volces.com/api/v3/images/generations"
SEEDREAM_KEY = "d155ace0-ee4d-42b1-936e-4a16d2623c89"
MODEL = "doubao-seedream-4-5-251128"

FILE_SERVER = "https://mjy.gzlex.com:8095/api/sse/file/upload"
FILE_TOKEN = "ampzOmhjenFAMTIz"

def generate_and_upload(prompt):
    """生成图片并上传到文件服务器，返回永久链接"""
    print(f"[Gen] {prompt[:50]}...")
    
    headers = {
        "Content-Type": "application/json",
        "Authorization": f"Bearer {SEEDREAM_KEY}"
    }
    
    data = {
        "model": MODEL,
        "prompt": prompt,
        "sequential_image_generation": "disabled",
        "response_format": "url",
        "watermark": False
    }
    
    resp = requests.post(SEEDREAM_URL, headers=headers, json=data, timeout=120)
    
    if resp.status_code != 200:
        print(f"[Err] HTTP {resp.status_code}: {resp.text[:100]}")
        return None
    
    result = resp.json()
    
    if "error" in result:
        print(f"[Err] API Error: {result.get('error', result)[:100]}")
        return None
    
    image_url = result["data"][0]["url"]
    print(f"[OK] Seedream URL: {image_url[:80]}...")
    
    # 下载到临时文件
    import tempfile
    import shutil
    
    temp_file = tempfile.NamedTemporaryFile(suffix='.jpg', delete=False)
    temp_path = temp_file.name
    temp_file.close()
    
    print(f"[Download] Saving to {temp_path}")
    
    # 使用 session 处理重定向
    session = requests.Session()
    img_resp = session.get(image_url, timeout=60)
    
    with open(temp_path, 'wb') as f:
        f.write(img_resp.content)
    
    print(f"[OK] Downloaded, size: {os.path.getsize(temp_path)} bytes")
    
    # 上传到文件服务器
    print(f"[Upload] Uploading to file server...")
    with open(temp_path, "rb") as f:
        files = {"file": f}
        upload_resp = requests.post(FILE_SERVER, headers={"Authorization": f"Bearer {FILE_TOKEN}", "User-Agent": "Apifox/1.0.0"}, files=files, timeout=60)
    
    # 清理临时文件
    os.remove(temp_path)
    
    result = upload_resp.json()
    
    if result.get("code") != "0":
        print(f"[Err] Upload failed: {result}")
        return None
    
    return result["data"]["fileUrl"]

def upload_file(file_path):
    """上传文件到文件服务器"""
    print(f"[Upload] {file_path}")
    
    headers = {
        "Authorization": f"Bearer {FILE_TOKEN}",
        "User-Agent": "Apifox/1.0.0"
    }
    
    with open(file_path, "rb") as f:
        files = {"file": f}
        resp = requests.post(FILE_SERVER, headers=headers, files=files, timeout=60)
    
    result = resp.json()
    
    if result.get("code") != "0":
        print(f"[Err] Upload failed: {result}")
        return None
    
    return result["data"]["fileUrl"]

def main():
    args = sys.argv[1:]
    
    if not args:
        print("""
HTML Generator - AI 生图脚本

Usage:
  python html-gen.py generate "prompt"     生成图片
  python html-gen.py upload /path/to/file  上传文件

Examples:
  python html-gen.py generate "A premium Chinese baijiu bottle"
  python html-gen.py upload /tmp/image.jpg
""")
        return
    
    cmd = args[0]
    
    if cmd == "generate":
        if len(args) < 2:
            print("Usage: python html-gen.py generate 'prompt'")
            return
            
        prompt = args[1]
        url = generate_and_upload(prompt)
        
        if url:
            print(f"[OK] Permanent URL: {url}")
            
    elif cmd == "upload":
        if len(args) < 2:
            print("Usage: python html-gen.py upload /path/to/file")
            return
            
        file_path = args[1]
        if not os.path.exists(file_path):
            print(f"[Err] File not found: {file_path}")
            return
            
        url = upload_file(file_path)
        
        if url:
            print(f"[OK] File URL: {url}")
            
    else:
        print(f"Unknown command: {cmd}")

if __name__ == "__main__":
    main()

File: skills/html-generator/scripts/html-gen.js
#!/usr/bin/env node

// ============================================================================
// HTML Generator - AI 生图核心脚本
// 调用 Seedream API 生成图片，上传到文件服务器
// ============================================================================

import { readFile, writeFile } from 'fs/promises';
import { execSync } from 'child_process';
import { tmpdir } from 'os';
import { join } from 'path';

// -- 配置 ---------------------------------------------------------------

const CONFIG = {
  // Seedream API
  SEEDREAM_URL: 'https://ark.cn-beijing.volces.com/api/v3/images/generations',
  SEEDREAM_KEY: 'd155ace0-ee4d-42b1-936e-4a16d2623c89',
  MODEL: 'doubao-seedream-5-0-260128',
  
  // 文件服务器
  FILE_SERVER: 'https://mjy.gzlex.com:8095/api/sse/file/upload',
  FILE_TOKEN: 'ampzOmhjenFAMTIz',
};

// -- 工具函数 -----------------------------------------------------------

function generateImage(prompt, size = '2048x2048') {
  console.log(`🎨 生成图片: ${prompt.substring(0, 50)}...`);
  
  const cmd = `curl -s -X POST "${CONFIG.SEEDREAM_URL}" \\
    -H "Content-Type: application/json" \\
    -H "Authorization: Bearer ${CONFIG.SEEDREAM_KEY}" \\
    -d '${JSON.stringify({
      model: CONFIG.MODEL,
      prompt: prompt,
      sequential_image_generation: 'disabled',
      response_format: 'url',
      size: size,
      watermark: false
    })}'`;
  
  const result = execSync(cmd, { encoding: 'utf-8' });
  const data = JSON.parse(result);
  
  if (data.error) {
    throw new Error(`Seedream API error: ${data.error.message}`);
  }
  
  return data.data[0].url;
}

function uploadToFileServer(filePath) {
  console.log(`📤 上传到文件服务器: ${filePath}`);
  
  const cmd = `curl -s -X POST "${CONFIG.FILE_SERVER}" \\
    -H "Authorization: Bearer ${CONFIG.FILE_TOKEN}" \\
    -H "User-Agent: Apifox/1.0.0" \\
    -F "file=@${filePath}"`;
  
  const result = execSync(cmd, { encoding: 'utf-8' });
  const data = JSON.parse(result);
  
  if (data.code !== '0') {
    throw new Error(`Upload failed: ${result}`);
  }
  
  return data.data.fileUrl;
}

function downloadImage(url, outputPath) {
  console.log(`⬇️ 下载图片: ${url}`);
  
  const cmd = `curl -s -o "${outputPath}" "${url}"`;
  execSync(cmd, { encoding: 'utf-8' });
  
  return outputPath;
}

// -- 主程序 -----------------------------------------------------------

async function main() {
  const args = process.argv.slice(2);
  const command = args[0];
  
  if (command === 'generate') {
    // 生成图片并上传
    const prompt = args[1] || args[--args];
    if (!prompt) {
      console.error('Usage: node html-gen.js generate "prompt text"');
      process.exit(1);
    }
    
    try {
      // 1. 生成图片
      const tempUrl = generateImage(prompt);
      console.log(`✅ Seedream: ${tempUrl}`);
      
      // 2. 下载到临时文件
      const tempPath = join(tmpdir(), `html-gen-${Date.now()}.jpg`);
      downloadImage(tempUrl, tempPath);
      
      // 3. 上传到文件服务器
      const permanentUrl = uploadToFileServer(tempPath);
      console.log(`✅ 永久链接: ${permanentUrl}`);
      
      console.log('\n📋 URL for HTML:');
      console.log(permanentUrl);
      
    } catch (err) {
      console.error('❌ Error:', err.message);
      process.exit(1);
    }
    
  } else if (command === 'upload') {
    // 仅上传文件
    const filePath = args[1];
    if (!filePath) {
      console.error('Usage: node html-gen.js upload /path/to/file');
      process.exit(1);
    }
    
    try {
      const url = uploadToFileServer(filePath);
      console.log(`✅ 永久链接: ${url}`);
    } catch (err) {
      console.error('❌ Error:', err.message);
      process.exit(1);
    }
    
  } else {
    console.log(`
HTML Generator - AI 生图脚本

Usage:
  node html-gen.js generate "prompt"     生成图片并上传
  node html-gen.js upload /path/to/file  仅上传文件

Examples:
  node html-gen.js generate "A premium Chinese baijiu bottle on dark table"
  node html-gen.js upload /tmp/image.jpg
`);
  }
}

main();
